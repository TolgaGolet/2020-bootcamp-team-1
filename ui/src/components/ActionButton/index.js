import React from "react";
import Button from "react-bootstrap/Button";
import Fab from "../common/Fab";
import LoadingIndicator from "../common/LoadingIndicator";

const ActionButton = ({ isLoggedIn, onClick, isPost, isLoading = false }) => {
  return (
    <div className="d-flex justify-content-center">
      <Button
        className={isLoggedIn ? "mr-3 hide-below-medium" : undefined}
        style={isPost ? { maxWidth: "40vw", marginBottom: "10vh" } : undefined}
        disabled={isLoading}
        block={isPost}
        onClick={onClick}
      >
        {isLoggedIn ? (
          isPost ? (
            isLoading ? (
              <LoadingIndicator />
            ) : (
              "Save"
            )
          ) : (
            "Create New Recipe"
          )
        ) : (
          "Join"
        )}
      </Button>
      {isLoggedIn && <Fab type={isPost ? "post" : "plus"} onClick={onClick} />}
    </div>
  );
};

export default ActionButton;
