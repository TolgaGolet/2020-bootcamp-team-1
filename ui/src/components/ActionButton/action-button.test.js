import React from "react";
import { shallow } from "enzyme";
import ActionButton from "./";

const ButtonComponent = "Button";
const LoadingIndicatorComponent = "LoadingIndicator";

const setup = (props = {}) => {
  const component = shallow(<ActionButton {...props} />);
  return component;
};

describe("ActionButton Component", () => {
  it("Renders correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders an action button without any prop", () => {
    const wrapper = setup();
    const div = wrapper.find("div");
    expect(div.length).toBe(1);
  });

  it("Should render text 'Join' when not logged in", () => {
    const wrapper = setup({ isLoggedIn: false });
    const text = wrapper.find(ButtonComponent).text();
    expect(text).toBe("Join");
  });

  it("Should render loading indicator when logged in and posting", () => {
    const wrapper = setup({ isLoggedIn: true, isLoading: true, isPost: true });
    const component = wrapper
      .find(ButtonComponent)
      .find(LoadingIndicatorComponent);
    expect(component).toBeDefined();
  });
});
