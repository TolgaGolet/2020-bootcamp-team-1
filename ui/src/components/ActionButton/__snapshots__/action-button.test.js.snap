// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton Component Renders correctly 1`] = `ShallowWrapper {}`;
